test/Microsoft.AspNet.Mvc.ModelBinding.Test/BodyBindingAndValidationIntegrationTest.cs

```diff
@@ -0,0 +1,186 @@
+
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Test
+{
+    public class BodyValidationIntegrationTests
+    {
+        private class Person
+        {
+            [FromBody]
+            [Required]
+            public Address Address { get; set; }
+        }
+
+        private class Address
+        {
+            public string Street { get; set; }
+        }
+
+        [Fact]
+        public async Task BodyBoundOnProperty_RequiredOnProperty_AddsModelStateError()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderModelName = "CustomParameter",
+                },
+                ParameterType = typeof(Person)
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
+            var httpContext = operationContext.HttpContext;
+            ConfigureHttpRequest(httpContext.Request, string.Empty);
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var model = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.Equal("The Address field is required.", modelState[""].Errors.Single().ErrorMessage);
+        }
+
+        private class Person4
+        {
+            [FromBody]
+            [Required]
+            public int Address { get; set; }
+        }
+
+        [Fact]
+        public async Task BodyBoundOnProperty_RequiredOnValueTypeProperty_AddsModelStateError()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderModelName = "CustomParameter",
+                },
+                ParameterType = typeof(Person4)
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
+            var httpContext = operationContext.HttpContext;
+            ConfigureHttpRequest(httpContext.Request, string.Empty);
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var model = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.StartsWith("No JSON content found and type 'System.Int32' is not nullable.",
+                modelState[""].Errors.Single().Exception.Message);
```

Comment:
Could you suggest what the model state key should be in this case. 
This exception originates from the JsonSerializer and we would need to change the design to get it to a desriable model state key ( if we choose to do something different ). 
This PR I am only ensuring we are consistent across the board.